pipelines/common/src/main/java/com/google/fhir/analytics/DwhFiles.java

```diff
@@ -186,59 +186,20 @@ public boolean hasIncrementalDir() throws IOException {
     return matchResult.status() == Status.OK;
   }
 
-  /**
-   * This method returns the list of non-empty materialized view directories which contains at least
-   * one file under it. When the Full Run creates view directories, the name is always lowercase.
-   * FHIR ResourceType directories are capitalized
-   *
-   * @return the set of non-empty view directories
-   * @throws IOException
-   */
-  public Set<String> findNonEmptyViewTypes() throws IOException {
-    // TODO : If the list of files under the dwhRoot is huge then there can be a lag in the api
-    //  response. This issue https://github.com/google/fhir-data-pipes/issues/288 helps in
-    //  maintaining the number of file to an optimum value.
-    String fileSeparator = getFileSeparatorForDwhFiles(dwhRoot);
-    List<MatchResult> matchedChildResultList =
-        FileSystems.match(
-            List.of(
-                getPathEndingWithFileSeparator(dwhRoot, fileSeparator)
-                    + "*"
-                    + fileSeparator
-                    + "*"));
-    Set<String> fileSet = new HashSet<>();
-    for (MatchResult matchResult : matchedChildResultList) {
-      if (matchResult.status() == Status.OK && !matchResult.metadata().isEmpty()) {
-        for (Metadata metadata : matchResult.metadata()) {
-          fileSet.add(metadata.resourceId().getCurrentDirectory().getFilename());
-        }
-      } else if (matchResult.status() == Status.ERROR) {
-        log.error("Error matching resource types under {} ", dwhRoot);
-        throw new IOException(String.format("Error matching resource types under %s", dwhRoot));
-      }
-    }
-
-    Set<String> typeSet = Sets.newHashSet();
-    for (String file : fileSet) {
-      if (file.toLowerCase().equals(file)) {
-        typeSet.add(file);
-      }
-    }
-    log.info("View types under {} are {}", dwhRoot, typeSet);
-    return typeSet;
-  }
-
   /**
    * This method returns the list of non-empty directories which contains at least one file under
-   * it. The directory name should be a valid FHIR resource type.
+   * it.
    *
+   * @param viewFlag If true, the views returned will be valid FHIR Resource Types. If false, the
+   *     views will be valid View Definitions
    * @return the set of non-empty directories
-   * @throws IOException
+   * @throws IOException if the directory does not contain a valid file type
    */
-  public Set<String> findNonEmptyFhirResourceTypes() throws IOException {
+  public Set<String> findNonEmptyViews(Boolean viewFlag) throws IOException {
```

Comment:
Any particular reason for using `Boolean` instead of `boolean`?

Reply:
Fixed typo, no need for `Boolean` object here.